src/OrleansAzureUtils/Storage/RemindersTableManager.cs

```diff
@@ -118,53 +118,59 @@ private RemindersTableManager(Guid serviceId, string deploymentId, string storag
+                        sEnd));
+                string query = TableQuery.CombineFilters(filterOnServiceIdStr, TableOperators.And, filterBetweenBeginAndEnd);
                 var queryResults = await ReadTableEntriesAndEtagsAsync(query);
                 return queryResults.ToList();
             }
 
             if (begin == end)
             {
-                
-                Expression<Func<ReminderTableEntry, bool>> query =
-                    e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                        && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0;
-
-                var queryResults = await ReadTableEntriesAndEtagsAsync(query);
+                var queryResults = await ReadTableEntriesAndEtagsAsync(filterOnServiceIdStr);
                 return queryResults.ToList();
             }
 
             // (begin > end)
-            Expression<Func<ReminderTableEntry, bool>> p1Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sBegin) > 0;
-            Expression<Func<ReminderTableEntry, bool>> p2Query =
-                e => String.Compare(e.PartitionKey, serviceIdStr + '_') > 0
-                    && String.Compare(e.PartitionKey, serviceIdStr + (char)('_' + 1)) <= 0
-                    && String.Compare(e.PartitionKey, sEnd) <= 0;
-
-            var p1 = ReadTableEntriesAndEtagsAsync(p1Query);
-            var p2 = ReadTableEntriesAndEtagsAsync(p2Query);
-
-            IEnumerable<Tuple<ReminderTableEntry, string>>[] arr = await Task.WhenAll(p1, p2);
-            return  arr[0].Concat(arr[1]).ToList();
+            string filterExtendBeginAndEnd = TableQuery.CombineFilters(
+                    TableQuery.GenerateFilterCondition(nameof(ReminderTableEntry.PartitionKey), QueryComparisons.GreaterThan, sBegin),
+                    TableOperators.Or,
+                    TableQuery.GenerateFilterCondition(nameof(ReminderTableEntry.PartitionKey), QueryComparisons.LessThanOrEqual,
+                        sEnd));
+            string query1 = TableQuery.CombineFilters(
+                filterOnServiceIdStr, TableOperators.And, TableQuery.GenerateFilterCondition(nameof(ReminderTableEntry.PartitionKey), QueryComparisons.GreaterThan, sBegin));
+            string query2 = TableQuery.CombineFilters(
+                filterOnServiceIdStr, TableOperators.And, TableQuery.GenerateFilterCondition(nameof(ReminderTableEntry.PartitionKey), QueryComparisons.LessThanOrEqual, sEnd));
+
+            var results1 = ReadTableEntriesAndEtagsAsync(query1);
+            var results2 = ReadTableEntriesAndEtagsAsync(query2);
+            IEnumerable<Tuple<ReminderTableEntry, string>>[] results = await Task.WhenAll(results1, results2);
+            return results[0].Concat(results[1]).ToList();
```

Comment:
Might be crazy, but I'm pretty sure this is wrong. It's equivalent to a SQL statement like:
```SQL
SELECT * WHERE ParitionKey > "ServiceId_" AND PartitionKey <= "<ServiceId>`" AND PartitionKey >= <LowerBound>
UNION
SELECT * WHERE ParitionKey > "ServiceId_" AND PartitionKey <= "<ServiceId>`" AND PartitionKey <= <UpperBound>​​​
```
Which is equivalent (ignoring duplicates) to:
```SQL
SELECT * WHERE ParitionKey >= "ServiceId_" AND PartitionKey <= "<ServiceId>`"
```